### PR TITLE
Fixes fallback if a singular message isn't translated

### LINKED ISSUE #16626: Translator always uses plural if singular is not translated.
### Description

Incase a singular message isn't translated it will fallback to plural.

**To reproduce have a translation like the following:** 
msgid "{0} voksen"
msgid_plural "{0} voksne"
msgstr[0] ""
msgstr[1] ""

Translator will always return "<count> voksne" as the singular is empty.

Can be reproduced like: `__dn('custom', '{0} voksen', '{0} voksne', 1, 1)`

### CakePHP Version

4.3.3

### PHP Version

8.1

### PR DESCRIPTION
**Pull request by Apisathan**
Closes [#16626](https://github.com/cakephp/cakephp/issues/16626)

### CODE REVIEW

**Comment by ADmad** [comment]
You need to update the failing test too. Looking at that test I agree the current behavior doesn't make sense and it should be changed.

File: src/I18n/Translator.php [hunk 1]
```diff
@@ -157,6 +157,11 @@ public function translate(string $key, array $tokensValues = []): string
 
         if ($message === '') {
             $message = $key;
+
+            //If singular haven't been translated, fallback to the key.
+            if (isset($tokensValues['_singular']) && $tokensValues['_count'] <= 1) {
```
Comment (garas) [inline src/I18n/Translator.php]:
```suggestion
            if (isset($tokensValues['_singular']) && $tokensValues['_count'] === 1) {
```

_0 years_ vs _0 year_ (assuming source language is English)

File: src/I18n/Translator.php [hunk 2]
```diff
@@ -157,6 +157,11 @@ public function translate(string $key, array $tokensValues = []): string
 
         if ($message === '') {
             $message = $key;
+
+            //If singular haven't been translated, fallback to the key.
```
Comment (markstory) [inline src/I18n/Translator.php]:
```suggestion
            // If singular haven't been translated, fallback to the key.
```

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**